index.js

```diff
@@ -138,16 +139,21 @@ chainMethods.forEach(method => {
 })
 
 Chain.prototype.end = function (callback) {
+  if (this._hasInvoked === true) {
+    throw new Error('The dispatch function has already been invoked')
+  }
   doInject(this.dispatch, this.option, callback)
+  this._hasInvoked = true
```

Comment:
Can you make the chain return itself here? So that we can:

```js
await chain.end()
```

Reply:
I agree, `end` should not return `this`, because you shouldn't be able to set more things once you send the request.
At the moment, this is valid:
```js
inject(dispatch)
  .get('/')                   // set the request method to GET, and request URL to '/'
  .headers({ foo: 'bar' })    // set the request headers
  .payload('request payload') // set the request payload
  .body('request body')       // alias for payload
  .query({ foo: 'bar' })      // set the query parameters
  .end((err, res) => {
    console.log(res.payload)
  })
  .query({ foo: 'bar' }
```

Reply:
If `end()` don't return `this`, then I think we should always set the callback function in `end()`, thus there is no need to store the promise in `end()`, which is requested in https://github.com/fastify/light-my-request/pull/53#discussion_r335347129

Reply:
Why we don't return `this._promise`?

In this way:
- we stop the chaining
- we can `const res = await chain.end()`

Reply:
@Eomm Great idea! I agree.